File: 102-binary-tree-level-order-traversal/102-binary-tree-level-order-traversal.cpp

class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        
               
        vector<vector<int>> ans;
        queue<TreeNode*> q;
        
        if (root) {
            q.push(root);
        }
        
        TreeNode* cur;
        
        while (!q.empty()) {
            
            int size = q.size();
            
            ans.push_back(vector<int>());
            
            for (int i = 0; i < size; ++i) {		// traverse nodes in the same level
                cur = q.front();
                q.pop();
                
                ans.back().push_back(cur->val);		// visit the root
                
                if (cur->left) {
                    q.push(cur->left);				// push left child to queue if it is not null
                }
                if (cur->right) {
                    q.push(cur->right);				// push right child to queue if it is not null
                }
            }
        }
        return ans;
    }
};  
  

File: 102-binary-tree-level-order-traversal/102-binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

from collections import deque

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        
        ans = []
        
        q = []
        
        if root:
            q.append(root)
            
        
        while len(q)!=0:
            
            n = len(q)
            ans.append([])
            
            for i in range(n):
                
                curr = q.pop(0)
                
                ans[-1].append(curr.val)
                
                if curr.left:
                    q.append(curr.left)
                
                if curr.right:
                    q.append(curr.right)
        
        return ans
            
            
        
        